encodings/alp/src/alp_rd/mod.rs

```diff
@@ -346,11 +346,12 @@ fn build_left_parts_dictionary<T: ALPRDFloat>(
     );
 
     // Count the number of occurrences of each left bit pattern
-    let counts = samples
+    let mut counts = HashMap::new();
```

Comment:
nice. did you find this in profiling or just noticed it? should we add it to our clippy as a disallowed-methods https://doc.rust-lang.org/clippy/lint_configuration.html#disallowed-methods ?